cpp/util/etcd.cc

```diff
@@ -108,44 +139,27 @@ void GetRequestDone(const string& keyname, EtcdClient::GetResponse* resp,
     return;
   }
 
-  const JsonObject node(*gen_resp->json_body, "node");
-  if (!node.Ok()) {
+  const JsonObject json_node(*gen_resp->json_body, "node");
+  if (!json_node.Ok()) {
     parent_task->Return(Status(util::error::FAILED_PRECONDITION,
                                "Invalid JSON: Couldn't find 'node'"));
     return;
   }
 
-  const JsonInt createdIndex(node, "createdIndex");
-  if (!createdIndex.Ok()) {
-    parent_task->Return(Status(util::error::FAILED_PRECONDITION,
-                               "Invalid JSON: Couldn't find 'createdIndex'"));
-    return;
-  }
-
-  const JsonInt modifiedIndex(node, "modifiedIndex");
-  if (!modifiedIndex.Ok()) {
-    parent_task->Return(Status(util::error::FAILED_PRECONDITION,
-                               "Invalid JSON: Couldn't find 'modifiedIndex'"));
+  StatusOr<EtcdClient::Node> node(ParseNodeFromJson(json_node));
+  if (!node.status().ok()) {
+    parent_task->Return(node.status());
     return;
   }
 
-  const JsonString key(node, "key");
-  if (!key.Ok()) {
-    parent_task->Return(Status(util::error::FAILED_PRECONDITION,
-                               "Invalid JSON: Couldn't find 'key'"));
-    return;
-  }
-
-  const JsonString value(node, "value");
-  if (!value.Ok()) {
+  if (node.ValueOrDie().deleted_) {
     parent_task->Return(Status(util::error::FAILED_PRECONDITION,
                                "Invalid JSON: Couldn't find 'value'"));
```

Comment:
Kinda reads a bit weirdly now with this error message for `if (node.ValueOrDie().deleted_)`, maybe update it too?